sympy/sets/handlers/intersection.py

```diff
@@ -80,14 +80,15 @@ def intersection_sets(a, b): # noqa:F811
 
 @dispatch(Range, Interval)  # type: ignore # noqa:F811
 def intersection_sets(a, b): # noqa:F811
-    from sympy.functions.elementary.integers import floor, ceiling
-    if not all(i.is_number for i in b.args[:2]):
+    # Check that there are no symbolic arguments
+    if not all(i.is_number for i in a.args + b.args[:2]):
```

Comment:
Should this be `if not any(i.is_number for i in a.args + b.args[2:])`?

Reply:
Maybe then the XFAIL would pass.

Reply:
The logic is that all the `Range` arguments and the two first `Interval` arguments should be numbers (the two last ones are booleans as I understand it). So rather than just testing that the `Interval` arguments are numbers, we also test that the `Range` arguments are numbers, as that caused the errors earlier. The tests fail as there is no code corresponding to #22083 that tests the start and end arguments separately and tries to make the most out of it. Yet.

At least based on how I interpreted the code.